src/observation_recipe.jl

```diff
@@ -0,0 +1,167 @@
+matrix. This is `(model_error_scale * seasonal_means).^2`, where
+`seasonal_means` is the seasonal mean for each of the season (DJF, MAM, JJA,
+SON).
+
+- `regularization`: A diagonal matrix of the form `regularization * I` is added
+to the covariance matrix.
+
+- `ignore_nan`: If `true`, then `NaN`s are ignored when computing the covariance
+matrix. Otherwise, `NaN` are included in the intermediate calculation of the
+covariance matrix. Note that all `NaN`s are removed in the last step of forming
+the covariance matrix even if `ignore_nan` is `false`.
+
+- `float_type`: The float type of the covariance matrix. The conversion to
+`float_type` is done just before returning the covariance matrix.
+"""
+function SeasonalDiagonalCovariance(;
+    model_error_scale = 0.0,
+    regularization = 0.0,
+    ignore_nan = true,
+    float_type = Float32,
+)
+    model_error_scale < zero(model_error_scale) &&
+        error("Model_error_scale ($model_error_scale) should not be negative")
+    regularization < zero(regularization) &&
+        error("Regularization ($regularization) should not be negative")
+
+    return SeasonalDiagonalCovariance(
+        model_error_scale,
+        regularization,
+        ignore_nan,
+        float_type,
+    )
+end
+
+"""
+    SVDplusDCovariance <: AbstractCovarianceEstimator
+
+Contain the necessary information to construct a `EKP.SVDplusD` covariance
+matrix from `ClimaAnalysis.OutputVar`s.
+"""
+struct SVDplusDCovariance{ # TODO: Should change the name of this to `SVDplusD`
+    FT1 <: AbstractFloat,
+    FT2 <: AbstractFloat,
+    DT <: DataType,
+    DATE <: Dates.AbstractDateTime,
+} <: AbstractCovarianceEstimator
+    """A model error scale term added to the diagonal of the covariance matrix"""
+    model_error_scale::FT1
+
+    """A regularization term added to the diagonal of the covariance matrix"""
+    regularization::FT2
+
+    """Float type of covariance matrix"""
+    float_type::DT
+
+    """Tuple of the start and end dates of the samples"""
+    sample_dates::Vector{NTuple{2, DATE}}
+end
+
+"""
+    SVDplusDCovariance(model_error_scale = 0.0,
+                       regularization = 0.0,
+                       float_type = Float32,
+                       sample_dates = nothing)
+
+Create a `SVDplusDCovariance` which specifies how the covariance matrix should
+be formed. When used with `ObservationRecipe.observation` or `ObservationRecipe.covariance`,
+return a `EKP.SVDplusD` covariance matrix.
+
+Positional arguments
+=====================
+
+- `sample_dates`: The start and end dates of each samples. This is used to
+determine the sample from the time series data of the `OutputVar`s. These dates
+must be present in all the `OutputVar`s.
+
+Keyword arguments
+=====================
+
+- `model_error_scale`: Noise from the model error added to the covariance
+matrix. This is `(model_error_scale * mean(samples, dims = 2)).^2`, where
+`mean(samples, dims = 2)` is the mean of the samples.
```

Comment:
Could you add a note surfacing the issue of not using years of data with `model_error_scale` ? You left a nice comment [here](https://github.com/CliMA/ClimaCalibrate.jl/pull/191/files#diff-c3d90bfa0e6a7f30b9ed65437de69858dcbe33ffb796b57f0abe308d963d1a59R178-R182) but I think it would be good to inform users as well.